app/components/OutcomeBar.tsx

```diff
@@ -1,64 +1,136 @@
-'use client';
-
-import { FixedProductMarketMaker } from '@/queries/omen';
 import { cx } from 'class-variance-authority';
+import { useQuery } from '@tanstack/react-query';
+import request from 'graphql-request';
+
+import { BlockDataType } from '@/queries/xdai';
+import {
+  FixedProductMarketMaker,
+  FpmmTrade_OrderBy,
+  getMarketTrades,
+} from '@/queries/omen';
+import { OMEN_SUBGRAPH_URL, XDAI_BLOCKS_SUBGRAPH_URL } from '@/constants';
 import { Outcome } from '@/entities';
 
 interface OutcomeBarProps {
   market: FixedProductMarketMaker;
 }
 
+type OutcomeTokenMarginalPricesResponse = {
+  [key: string]: {
+    outcomeTokenMarginalPrices: string[];
+  };
+};
+
 export const OutcomeBar = ({ market }: OutcomeBarProps) => {
+  const { id } = market;
+
+  const { data: trades } = useQuery({
+    queryKey: ['getMarketTrades', id],
+    queryFn: async () =>
+      getMarketTrades({
+        fpmm: id,
+        orderBy: FpmmTrade_OrderBy.CreationTimestamp,
+      }),
```

Comment:
Let's fetch only one (the last) market trade.
If it's not the last one to be returned, use a different order direction.

```suggestion
    queryKey: ['getLastMarketTrade', id],
    queryFn: async () =>
      getMarketTrades({
        first: 1,
        fpmm: id,
        orderBy: FpmmTrade_OrderBy.CreationTimestamp,
      }),
```

Reply:
Thanks for the suggestion! Renamed the variable and fixed the `getMarketTrades` fn to accept the `first` parameter!